test/functional/eval/execute_spec.lua

```diff
@@ -62,30 +62,87 @@ describe('execute()', function()
-    feed(':let g:mes = execute("echon 42")<CR>')
-    screen:expect([[
-    ^                                        |
-    {0:~                                       }|
-    {0:~                                       }|
-    {0:~                                       }|
-    :let g:mes = execute("echon 42")        |
-    ]])
-    eq('42', eval('g:mes'))
+  describe('{silent} argument', function()
+    it('captures & displays output for ""', function()
+      local screen = Screen.new(40, 5)
+      screen:attach()
+      command('let g:mes = execute("echon 42", "")')
+      screen:expect([[
+      ^                                        |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+      42                                      |
+      ]])
+      eq('42', eval('g:mes'))
+    end)
+
+    it('captures but does not display output for "silent"', function()
+      local screen = Screen.new(40, 5)
+      screen:attach()
+      command('let g:mes = execute("echon 42")')
+      screen:expect([[
+      ^                                        |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+                                              |
+      ]])
+      eq('42', eval('g:mes'))
+
+      command('let g:mes = execute("echon 13", "silent")')
+      screen:expect([[
+      ^                                        |
+      ~                                       |
+      ~                                       |
+      ~                                       |
+                                              |
+      ]])
+      eq('13', eval('g:mes'))
+    end)
+
+    it('suppresses errors for "silent!"', function()
+      eq(0, exc_exec('let g:mes = execute(0.0, "silent!")'))
+      eq('', eval('g:mes'))
+
+      eq(0, exc_exec('let g:mes = execute("echon add(1, 1)", "silent!")'))
+      eq('1', eval('g:mes'))
+
+      eq(0, exc_exec('let g:mes = execute(["echon 42", "echon add(1, 1)"], "silent!")'))
+      eq('421', eval('g:mes'))
+    end)
+
+    it('propagates errors for "" and "silent"', function()
+      local ret
+      ret = exc_exec('call execute(0.0, "")')
+      eq('Vim(call):E806: using Float as a String', ret)
+
+      ret = exc_exec('call execute(v:_null_dict, "silent")')
+      eq('Vim(call):E731: using Dictionary as a String', ret)
+
+      ret = exc_exec('call execute("echo add(1, 1)", "")')
+      eq('Vim(echo):E714: List required', ret)
+
+      ret = exc_exec('call execute(["echon 42", "echo add(1, 1)"], "")')
+      eq('Vim(echo):E714: List required', ret)
+
+      ret = exc_exec('call execute("echo add(1, 1)", "silent")')
+      eq('Vim(echo):E714: List required', ret)
+
+      ret = exc_exec('call execute(["echon 42", "echo add(1, 1)"], "silent")')
+      eq('Vim(echo):E714: List required', ret)
```

Comment:
Oh, except that running inside of a `try/catch` prevents assignment to the variable.